.github/workflows/CI.yml

```diff
@@ -88,13 +88,17 @@ jobs:
     runs-on: ubuntu-latest
     container:
       image: ghcr.io/${{ github.repository_owner }}/chapel-github-ci:latest
+      options: --user root
       credentials:
         username: ${{ github.actor }}
         password: ${{ secrets.github_token }}
     steps:
     - uses: actions/checkout@v3
       with:
         fetch-depth: 100000
+    - name: Set ownership
+      run: |
+        chown -R $(id -u):$(id -g) $PWD   
```

Comment:
I don't really understand the issue well enough to know if this is an appropriate solution. But just based on the diff:
 - Do we need to `chown` here and later on? 
 - Should we be doing this for all jobs since it could impact any job doing git operations in the future? 
 - Why do we need root? Is the core "fix" to chown and we just need to be root in order to chown? Any potential security issues from running as root?

Reply:
> I don't really understand the issue well enough to know if this is an appropriate solution. But just based on the diff:
> 
> * Do we need to `chown` here and later on?
No, adding it in a step right after we checkout files is good enough. All the steps following that can make use of it.
> * Should we be doing this for all jobs since it could impact any job doing git operations in the future?
If the scripts that we are trying to run in the job/step is trying to access a file system that requires sudo access then we will have to add this to that specific job. Since we are not seeing any issues with other jobs in this workflow we will just fix this check_annotations_rt_calls job.
> * Why do we need root? Is the core "fix" to chown and we just need to be root in order to chown? Any potential security issues from running as root?
I removed root user added in the following PR just added chown. Since the change is added only to this specific job and git actor(git actor's access is changed only for this container) to run a script inside a container I do not see any security issue.

Reply:
>> Should we be doing this for all jobs since it could impact any job doing git operations in the future?

> If the scripts that we are trying to run in the job/step is trying to access a file system that requires sudo access then we will have to add this to that specific job. Since we are not seeing any issues with other jobs in this workflow we will just fix this check_annotations_rt_calls job.

My concern is that this issue went unnoticed, so it's possible other failure modes might be going unnoticed too. And I think @arezaii hit this when trying to add some new code to CI testing and I think it'd take future developers a while to jump from the error message to the solution required. I'd probably rather just fix for all jobs even if we don't see any failures in them currently.